modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -80,7 +80,7 @@ interface DomParser {
   addNodeFilter: (name: string, callback: ParserFilterCallback) => void;
   getNodeFilters: () => ParserFilter[];
   removeNodeFilter: (name: string, callback?: ParserFilterCallback) => void;
-  parse: (html: string, args?: ParserArgs) => AstNode;
+  parse: (html: string, args?: ParserArgs, isInsert?: boolean) => AstNode;
```

Comment:
I don't think we need to track inserts as separate things and if we need to it should really be a `ParserArgs` property in fact there seems to already be a `insert` boolean there might not be typed properly. The multiple headings in a summary is never valid regardless when you parse so I really think we should avoid making it a special thing for insert. If the editor some how manages to produce a incorrect structure by some other operation than insert then that will not be filtered up if we have this extra check in there.

- [ ] Remove extra insert tracking

Reply:
If I use `ParserArgs` `insert` property for that purpose InsertContetTest.ts would fail, due to [SetContentString](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/content/SetContentImpl.ts#L63) use it while set (not insert) content.

Reply:
When we have `h1` element as the first child of `summary] on `setContent` it is a valid child whereas on `inserContent` it is an invalid child and we unwrap it.

Reply:
It might be better that the single child rule is just a filter that you add. So the schema would make any h1 valid but the filter does a separate pass after the parsing has been done. That way you wouldn't need to modify the parser at all.

Reply:
No wait a simple filter would not work since you want to avoid the splitting. We just need to figure out where to add this, adding an extra insert tracking though the whole parser makes the entire thing more complex. You could in theory have a separate cleanup step in the insert code similar to how transparent elements are handled.

Reply:
Ok played around with this a bit. What you could do is make sure that it always parses with a full context if you insert headings into a summary element and do that in the InsertContentImpl. The problem here now is that the first child heading rule is not going to get applied if you paste simple content into a simple context since it tries to optimize that case in the insert content code. But this is a complex case so with a special extra check in the insert code you can force it to always take the full context code path and then the first child rule always gets applied properly since then it does it on a bigger fragment including the summary parent context and it's siblings.